### DISCUSSION TITLE
File-scoped class

### DISCUSSION THREAD

**Discussion by lucasteles** [dotnet/csharplang#6313] [Language Ideas] [answered]
It is not uncommon to have lots of single class files in C#, so with the same motivations of [file-scoped namespaces](https://github.com/dotnet/csharplang/blob/3c8559f186d4c5df5d1299b0eaa4e139ae130ab6/proposals/csharp-10.0/file-scoped-namespaces.md)  I think we could consider it for classes too. 

It would make a lot of files without another indentation level and give the feel of the top level statements outside Program.cs

Eg: 

this code 
```csharp
namespace Example;

public class Foo { 
    
    readonly string bar;
    public int Id { get; set; }
    
    public Foo (string bar)
    {
        this.bar = bar;
    }

    public void DoSomething()
    {
        Console.WriteLine("things");
    }

    public class InsideFoo { }

    internal static bool IsThisNice() => true;
}

```
Could be rewritten as 

```csharp
namespace Example;
public class Foo; 

readonly string bar;
public int Id { get; set; }

public Foo (string bar)
{
    this.bar = bar;
}

public void DoSomething()
{
    Console.WriteLine("things");
}

public class InsideFoo { }

internal static bool IsThisNice() => true;
```
maybe we could for this type of definition join the class with the namespace, so we could have: 
```csharp
namespace public class Example.Foo;
```

of course, we would be able to have only one namespace class per file

**Comment by alexrp** [comment, marked as answer]
#928
